truffleruby/src/main/java/org/truffleruby/extra/ffi/PointerNodes.java

```diff
@@ -150,10 +150,29 @@ public boolean setAutorelease(DynamicObject pointer, boolean autorelease) {
 
     }
 
-    @Primitive(name = "pointer_set_at_offset", lowerFixnum = { 1, 2 })
+    @Primitive(name = "pointer_set_at_offset", lowerFixnum = { 1, 2, 3 })
     @ImportStatic(RubiniusTypes.class)
     public static abstract class PointerSetAtOffsetPrimitiveNode extends PrimitiveArrayArgumentsNode {
 
+        @Specialization(guards = "type == TYPE_CHAR")
+        public long setAtOffsetChar(DynamicObject pointer, int offset, int type, int value) {
+            Layouts.POINTER.getPointer(pointer).putByte(offset, (byte) (value & 0xFF));
```

Comment:
What do you think would be correct or more clear?
Having tests for this would be awesome, maybe there are some in the ffi gem.